### PR TITLE
Do not insert fields that do not exist in the DB

### PR DESCRIPTION
**Pull request by aschempp**
I have a field in my DCA that has a default value but does not exist in the database. DC_Table tries to add that field value when creating a new record.

### CODE REVIEW

**Comment by fritzmg** [comment]
Typically you have to enable `doNotSaveEmpty` for that in your eval and return `null` in the `save_callback` on that field.

**Comment by aschempp** [comment]
this is not about _saving_ a record, but creating the initial one, which uses the `default` DCA option

**Comment by m-vo** [comment]
With "does not exist in the database" you mean the column does not exist, i.e. the migration did not run, yet?

**Comment by aschempp** [comment]
the column does not exist, because thats a virtual field and not used in any back end palette. In my case it is only there for the front end form that is based upon the DCA fields.

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 1]
```diff
@@ -615,11 +615,13 @@ public function create($set=array())
...
 		}
 
+		$databaseFields = array_column($this->Database->listFields($this->strTable), 'name');
+
 		// Get all default values for the new entry
 		foreach ($GLOBALS['TL_DCA'][$this->strTable]['fields'] as $k=>$v)
 		{
 			// Use array_key_exists here (see #5252)
-			if (\array_key_exists('default', $v))
+			if (\array_key_exists('default', $v) && \in_array($k, $databaseFields, true))
```
Comment (ausi) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: Do we need to compare the columns in lowercase?

Comment (fritzmg) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: Yes, `$this->Database->listFields($this->strTable)` will return the column names in lower case.

**Review by fritzmg — DISMISSED** [review]

File: core-bundle/src/Resources/contao/drivers/DC_Table.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: I don't think so. Doctrine does, but our method does not.

Comment (ausi) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: Would the case match if you use entities as well?

I’d prefer to compare them in lowercase to be sure.

Comment (fritzmg) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: Yeah, you are right - our method does not return it in lower case, my bad.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: Fact is that most DCAs use lowercase keys, so for this to work, the column names must also be lowercase, right?

Comment (ausi) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: We would need to lowercase them both, `$databaseFields` as well as `$k`, to be sure that we compare correctly IMO.

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:624]: I have no idea what you're talking about. The DCA uses the exact same column names as the database, we never lowercase that anywhere. Or should we lowercase the field names when writing to the database as well???

**Review by ausi — APPROVED** [review]

**Pull request merged**